### PR TITLE
pylint 0

### PR DESCRIPTION
**Pull request by mr-c**
Bumps Pylint score from 6.67/10 to 9.86/10

### CODE REVIEW

**Comment by mr-c** [comment]
Jenkins, retest this please

**Comment by ctb** [comment]
Did we discuss removing `collect_high_abundance_kmers` and `sandbox/find-high-abund-kmers.py` anywhere, in person or in an issue?  Ahh -- #1142. Please link in related issues when you do PRs, folks.

**Comment by mr-c** [comment]
Yep, was going to fill that out when I asked for review

On Sat, Aug 15, 2015, 10:26 C. Titus Brown <notifications@github.com> wrote:

> Did we discuss removing collect_high_abundance_kmers and
> sandbox/find-high-abund-kmers.py anywhere, in person or in an issue? Ahh
> -- #1142 <https://github.com/dib-lab/khmer/issues/1142>. Please link in
> related issues when you do PRs, folks.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dib-lab/khmer/pull/1255#issuecomment-131402974>.
>
-- 
Michael R. Crusoe: Programmer & Bioinformatician crusoe@ucdavis.edu
The lab for Data Intensive Biology; University of California, Davis
https://impactstory.org/MichaelRCrusoe http://twitter.com/biocrusoe

**Comment by mr-c** [comment]
@camillescott @ctb ready for review & merge

File: khmer/khmer_args.py [hunk 1]
```diff
@@ -1,16 +1,14 @@
-#
 # vim: set encoding=utf-8
 # This file is part of khmer, https://github.com/dib-lab/khmer/, and is
 # Copyright (C) Michigan State University, 2014-2015. It is licensed under
 # the three-clause BSD license; see LICENSE.
 # Contact: khmer-project@idyll.org
-#
+"""Common ArgParse constructs."""
```
Comment (ctb) [inline khmer/khmer_args.py]: Lowercase 'argparse' pls - https://docs.python.org/2/library/argparse.html

File: khmer/khmer_args.py [hunk 2]
```diff
@@ -55,9 +70,9 @@ def optimal_size(num_kmers, mem_cap=None, fp_rate=None):
       - fp_rate: the desired false positive rate [optional, conflicts with M]
     """
     if all((num_kmers is not None, mem_cap is not None, fp_rate is None)):
-        return estimate_optimal_with_K_and_M(num_kmers, mem_cap)
+        return estimate_optimal_with_k_and_m(num_kmers, mem_cap)
```
Comment (ctb) [inline khmer/khmer_args.py]: why lowercase 'M'? I am -0 on this, for two reasons --

- `-M` is the command line option we're using!
- mathematically, `M` and `N` are things we (and others) use in equations, and they are typically uppercase.
- well, and I see no point to this change.

File: lib/Makefile [hunk 3]
```diff
@@ -254,7 +254,7 @@ oxli.pc: oxli.pc.in
 	sed -e 's,@prefix@,$(PREFIX),'  -e 's,@VERSION@,$(VERSION),' $< >$@
 
 $(ZLIB_OBJS):
-	(cd $(ZLIB_DIR) && ./configure && make $(ZLIB_OBJS_BASE))
+	(cd $(ZLIB_DIR) && ./configure && make -f Makefile.pic $(ZLIB_OBJS_BASE))
```
Comment (ctb) [inline lib/Makefile]: why this? (no objections, just rather leary of stuff that changes semantics along with syntax in the same large PR, at this point.) Is this going to break a bunch of installs?

File: oxli/partition.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+#! /usr/bin/env python
```
Comment (ctb) [inline oxli/partition.py]: Please move this change to a new PR.

Comment (ctb) [inline oxli/partition.py]: (Explanation: if you're fixing syntax and deleting things, fine; but this kind of bigger rearrangement needs to be thought through separately from all the other changes.)

File: scripts/do-partition.py [hunk 5]
```diff
@@ -27,8 +26,7 @@
                               add_threading_args, sanitize_epilog)
 import glob
 from khmer.kfile import check_input_files, check_space
-import re
-import platform
+from oxli.partition import worker
```
Comment (ctb) [inline scripts/do-partition.py]: this is also too generic a name, I think (`worker`)

File: tests/test_scripts.py [hunk 6]
```diff
@@ -1079,66 +1075,39 @@ def _make_graph(infilename, min_hashsize=1e7, n_hashes=2, ksize=20,
     return outfile
 
 
-def _DEBUG_make_graph(infilename, min_hashsize=1e7, n_hashes=2, ksize=20,
```
Comment (ctb) [inline tests/test_scripts.py]: Sorry, could you explain this change/function removal?

File: tests/test_scripts.py [hunk 7]
```diff
@@ -1079,66 +1075,39 @@ def _make_graph(infilename, min_hashsize=1e7, n_hashes=2, ksize=20,
...
+    ht.load_tagset(graphbase + '.tagset')
+    ht.load_partitionmap(final_pmap_file)
 
-    return outfile
+    x = ht.count_partitions()
+    assert x == (1, 0), x          # should be exactly one partition.
 
 
-def test_partition_graph_1():
+def test_partition_graph_too_many_threads():
```
Comment (ctb) [inline tests/test_scripts.py]: Please move to a new PR.

File: tests/test_scripts.py [hunk 8]
```diff
@@ -1717,6 +1686,25 @@ def test_do_partition():
     assert len(parts) == 1
 
 
+def test_do_partition_no_big_traverse():
```
Comment (ctb) [inline tests/test_scripts.py]: Please move to a new PR.

File: tests/test_scripts.py [hunk 9]
```diff
@@ -2006,6 +1994,22 @@ def test_make_initial_stoptags_load_stoptags():
     assert os.path.exists(outfile1), outfile1
 
 
+def test_extract_paired_reads_unpaired():
```
Comment (ctb) [inline tests/test_scripts.py]: Please move to a new PR with some explanation/motivation/??

**Comment by ctb** [comment]
Overall commentary --

* the pylint changes look great!
* too many additional things (functions removal, refactoring, new tests) mixed in with a bajllion syntax changes. On cursory inspection the bigger changes mostly look OK but I can't sort them out from the background. Move to new PR/individually to motivate, discuss.
* Note, I see no reason to have the refactoring block 2.0. Am I missing something?

File: khmer/khmer_args.py (hunk 2 above)
Comment (mr-c) [inline khmer/khmer_args.py]: pylint didn't like the name; I can add an override.

File: tests/test_scripts.py (hunk 6 above)
Comment (mr-c) [inline tests/test_scripts.py]: It is unused; `ipdb` provides a better debugging interface (just drop `import ipdb; ipdb.set_trace()` into the code in question) without mucking up the code.

File: tests/test_scripts.py (hunk 9 above)
Comment (mr-c) [inline tests/test_scripts.py]: sure; it is for test coverage.

File: tests/test_scripts.py (hunk 6 above)
Comment (ctb) [inline tests/test_scripts.py]: ok.

File: khmer/khmer_args.py (hunk 2 above)
Comment (mr-c) [inline khmer/khmer_args.py]: Fixed

File: lib/Makefile (hunk 3 above)
Comment (mr-c) [inline lib/Makefile]: removed

File: oxli/partition.py (hunk 4 above)
Comment (mr-c) [inline oxli/partition.py]: Done, see #1278 & #1277

File: khmer/khmer_logger.py [hunk 10]
```diff
@@ -34,9 +31,8 @@ def log_error(message, **kwargs):
     print(message, file=sys.stderr)
 
 
-def log_debug(message, **kwagrs):
+def log_debug(message, **kwargs):
```
Comment (mr-c) [inline khmer/khmer_logger.py]: This needs moving to its own PR

File: khmer/thread_utils.py [hunk 11]
```diff
@@ -12,8 +12,7 @@
 import threading
 import sys
 import screed
-from khmer import utils
-from khmer.utils import write_record
+from khmer.utils import write_record, check_is_pair
```
Comment (mr-c) [inline khmer/thread_utils.py]: This needs moving to its own PR

File: setup.py [hunk 12]
```diff
@@ -169,15 +169,31 @@ def check_for_openmp():
...
+                  "Kaben Nanlohy, Alexander Johan Nederbragt, "
+                  "Humberto Ortiz-Zuazaga, Jeramia Ory, Jason Pell, "
+                  "Charles Pepe-Ranney, Zachary N Russ, Erich Schwarz, "
+                  "Camille Scott, Josiah Seaman, Scott Sievert, "
+                  "Jared Simpson, Connor T. Skennerton, James Spencer, "
+                  "Ramakrishnan Srinivasan, Daniel Standage, "
+                  "James A. Stapleton, Joe Stein, Susan R Steinman, "
+                  "Benjamin Taylor, Will Trimble, Heather L. Wiencko, "
+                  "Michael Wright, Brian Wyss, Qingpeng Zhang, en zyme, "
+                  "C. Titus Brown",
```
Comment (mr-c) [inline setup.py]: This needs moving to its own PR

File: khmer/thread_utils.py (hunk 11 above)
Comment (mr-c) [inline khmer/thread_utils.py]: See #1284

**Comment by mr-c** [comment]
Jenkins, retest this please

**Comment by mr-c** [comment]
PR moved to '2.0+' milestone. Several pull requests have been spun out from this, only pylint/cppcheck fixes remain.

**Comment by mr-c** [comment]
Ready for review & merge!

File: khmer/khmer_logger.py (hunk 10 above)
Comment (ctb) [inline khmer/khmer_logger.py]: bump.

**Comment by ctb** [comment]
Apart from my one bump, looks great. @luizirber any comments?

**Comment by luizirber** [comment]
LGTM

**Comment by ctb** [comment]
great! you want to push the magic shiny button?

**Comment by luizirber** [comment]
Needs fixing (merge conflicts). I will check.

**Comment by mr-c** [comment]
Thanks!

On Mon, May 9, 2016 at 8:35 PM, Luiz Irber <notifications@github.com> wrote:

> Merged #1255 <https://github.com/dib-lab/khmer/pull/1255>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/dib-lab/khmer/pull/1255#event-654780142>
>



-- 
Michael R. Crusoe                          michael.crusoe@gmail.com
Community Engineer                 Common Workflow Language project
https://impactstory.org/u/0000-0002-2961-9670   +32 (0) 2 808 25 58
                                                    +1 480 627 9108

**Comment by luizirber** [comment]
@mr-c thank you, great work!
